examples/gaussian_process/plot_gpr_noisy_targets.py

```diff
@@ -11,104 +11,143 @@
 In both cases, the kernel's parameters are estimated using the maximum
 likelihood principle.
 
-The figures illustrate the interpolating property of the Gaussian Process
-model as well as its probabilistic nature in the form of a pointwise 95%
-confidence interval.
+The figures illustrate the interpolating property of the Gaussian Process model
+as well as its probabilistic nature in the form of a pointwise 95% confidence
+interval.
 
-Note that the parameter ``alpha`` is applied as a Tikhonov
-regularization of the assumed covariance between the training points.
+Note that the parameter `alpha` is applied as a Tikhonov regularization ridge
+parameter on the assumed covariance between the training points.
 """
 print(__doc__)
 
 # Author: Vincent Dubourg <vincent.dubourg@gmail.com>
 #         Jake Vanderplas <vanderplas@astro.washington.edu>
-#         Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>s
+#         Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#         Guillaume Lemaitre <g.lemaitre58@gmail.com>
 # License: BSD 3 clause
 
+# %%
+# Dataset generation
+# ------------------
+#
+# We will start by generating a synthetic dataset. The true generative process
+# is defined as :math:`f(x) = x \sin(x)`.
 import numpy as np
-from matplotlib import pyplot as plt
 
+X = np.linspace(start=0, stop=10, num=1_000).reshape(-1, 1)
+y = np.squeeze(X * np.sin(X))
+
+# %%
+import matplotlib.pyplot as plt
+
+plt.plot(X, y, label=r"$f(x) = x \sin(x)$", linestyle="dotted")
+plt.legend()
+plt.xlabel("$x$")
+plt.ylabel("$f(x)$")
+_ = plt.title("True generative process")
+
+# %%
+# We will use this dataset in the next experiment to illustrate how Gaussian
+# process regression is working.
+#
+# Example with noise-free target
+# ------------------------------
+#
+# In this first example, we will use the true generative process without
+# adding any noise. For training the Gaussian process regression, we will only
+# select few samples.
```

Comment:
Is capitalization better for consistency?

This suggestion also applies bellow.
```suggestion
# %%
# We will use this dataset in the next experiment to illustrate how Gaussian
# Process regression is working.
#
# Example with noise-free target
# ------------------------------
#
# In this first example, we will use the true generative process without
# adding any noise. For training the Gaussian Process regression, we will only
# select few samples.
```